scripts/extensions/videoEditor/src/VideoEditor.tsx

```diff
@@ -0,0 +1,688 @@
+            let refValue = this.videoRef.current!.getBoundingClientRect();
+            let currentValue = this.state.degree % 180 === -90 ? refValue.width : refValue.height;
+            let delta = (scale / crop.scale!) * (currentValue / crop.value!);
+
+            crop = {
+                ...crop,
+                aspect: 1 / crop.aspect!,
+                x: crop.y! * delta,
+                y: currentValue! - (crop.x! + crop.width!) * delta,
+                height: crop.width! * delta,
+                width: crop.height! * delta,
+                scale: scale,
+                value: currentValue,
+            };
+
+            const cropRef = this.reactCropRef.current?.['componentRef'];
+
+            if (cropRef) {
+                cropRef.style.visibility = 'unset';
+            }
+        }
+        this.setState({degree: degree, crop: crop, scale: scale}, this.checkIsDirty);
+    }
+
+    handleCrop = (newCrop: IVideoEditor['crop']) => {
+        if (
+            newCrop.x == null
+            || newCrop.y == null
+            || newCrop.width == null
+            || newCrop.height == null
+        ) {
+            throw new Error('Invalid state');
+        }
+
+        newCrop.x = Math.floor(newCrop.x);
+        newCrop.y = Math.floor(newCrop.y);
+        newCrop.width = Math.floor(newCrop.width);
+        newCrop.height = Math.floor(newCrop.height);
+
+        // newCrop lost aspect when cropping while rotating sometimes
+        if (!newCrop.aspect) {
+            newCrop.aspect = this.state.crop.aspect;
+        }
+
+        let refValue = this.videoRef.current!.getBoundingClientRect();
+
+        const crop = {
+            ...newCrop,
+            scale: this.getScale(),
+            value: this.state.degree % 180 === -90 ? refValue.width : refValue.height,
+        };
+
+        // when first draw crop zone, ReactImageCrop trigger a bulk of change event with the same
+        // newCrop value, using throttle with value about 50 did not help much but increase interval \
+        // may result in lagging
+        if (Object.values(this.state.crop).toString() === Object.values(crop).toString()) {
+            return;
+        }
+
+        this.setState({crop: crop}, this.checkIsDirty);
+    }
+
+    handleToggleVideo = () => {
+        if (this.state.playing) {
+            this.videoRef.current!.pause();
+        } else {
+            this.videoRef.current!.play();
+        }
+    }
+
+    handleToggleCrop = (aspect: number) => {
+        const cropAspect = aspect || this.state.crop.aspect || 0;
+        let crop = this.initState.crop;
+
+        if (this.state.cropEnabled === false) {
+            crop = this.getCropSample(cropAspect);
+        }
+        this.reactCropMarginDelta = 0;
+        this.setState({cropEnabled: !this.state.cropEnabled, crop: crop}, () => {
+            this.checkIsDirty();
+            // chrome adds 1rem extra (ghost) margin to ReactCrop cause crop area and video mismatch
```

Comment:
We think it's Chrome's fault, as you can see, [Firefox](https://user-images.githubusercontent.com/45254285/70311265-e6aa0900-1843-11ea-8fd9-552da267ba7f.png) does not need subtract margin for Video and React Crop to match but [Chrome does](https://user-images.githubusercontent.com/45254285/70311295-f9244280-1843-11ea-8951-5fd52ed4e704.png) (because it added [some extra position](https://user-images.githubusercontent.com/45254285/70312301-f0cd0700-1845-11ea-8a5d-1231a9967829.png)), if we remove margin for both, while it can reduce complex calculation, the [video is right next](https://user-images.githubusercontent.com/45254285/70311605-88315a80-1844-11ea-892a-4a7623791e29.png) to confirm bar, make it look a bit unnatural.

Should we remove margin for both video and react crop or look for other simpler solution?